README.md

```diff
@@ -2,19 +2,56 @@
 
 ## Description
 
-TODO: Describe your charm in a few paragraphs of Markdown
+This charm is used to configure an Ory Kratos charm to use an external provider.
 
 ## Usage
 
-TODO: Provide high-level usage, such as required config or relations
+### Client registration
 
-## Relations
+Before deploying this charm you should go and register an oidc client with the provider you wish to use. Instructions for
+registering a client for some well known provider can be found below. Instructions for a larger list of providers can be found at the
+Ory Kratos [docs](https://www.ory.sh/docs/kratos/social-signin/overview)
 
-TODO: Provide any relations which are provided or required by your charm
+Note that after registering a client you will need to provide a redirect_uri to the provider, you will get the redirect_uri
+after deploying the integrator.
 
-## OCI Images
+#### Azure AD
 
-TODO: Include a link to the default image your charm uses
+Instructions for registering a client on Azure AD can be found [here](https://learn.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app).
+
+#### Okta
+
+Instructions for registering a client on Azure AD can be found [here](https://developer.okta.com/docs/guides/find-your-app-credentials/main/).
+
+### Deployment
+
+For the kratos-external-idp-integrator charm to operate you need to deloy it, configure it and relate to a kratos charm.:
+```commandline
+juju deploy kratos-external-provider-integrator
+juju config kratos-external-provider-integrator ...
```

Comment:
I wasn't sure if documenting the configurations here was necessary, because charmhub will autogenerate docs about the config using `config.yaml`. I looked into other charms and it seems like most of them don't do this.

However if you feel like adding a couple of words about the configurations will make things clearer let me know.

Reply:
I wasn't aware of the autogenerated docs. It would be nice to have info about required configs or link to the charmhub documentation